Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift

```diff
@@ -50,7 +50,24 @@ internal extension String {
     }
 
     var isValidCardNumber: Bool {
-        return count >= 13 && count <= 19 && isValidLuhn
+        let clearedCardNumber = self.withoutNonNumericCharacters
+        
+        if !(clearedCardNumber.count >= 13 && clearedCardNumber.count <= 19 && clearedCardNumber.isValidLuhn) {
+            return false
+        }
```

Comment:
[Feedback]
I believe you don't need to check the length anymore, it's already checked below.
Similarly, you should do the luhn check after the length is checked: it doesn't make to consider a card non-luhn valid if it's not the right size.

Reply:
Done